### PR TITLE
Apply `clip_timerange` to time dependent fx variables

### LINKED ISSUE #1602: `clip_timerange` not applied to time dependent fx variables
**Describe the bug**
As mentioned somewhere in #1133 , `clip_timerange` is not being applied to time dependent fx variables. This can lead to cases in which a fx variable will not be added to the variable cube because the shapes between cubes will not match, since the variable cube will have been clipped in a previous step.

### PR DESCRIPTION
**Pull request by sloosvel**
## Description

This PR makes sure to clip the timerange in time dependent fx variables

Closes #1602 

Link to documentation:

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by zklaus** [comment]
We really have an unhealthy confusion in terminology. Outside of ESMValTool, fx stands for fixed in time. It makes no sense to talk of time-dependent fx files or variables. In ESMValTool we are conflating a few things, three, I think. Those are time-independent things associated with fx, cell measures that give information about grid cells, and ancillary variables that give extra information related to other variables, for example, a quality flag or a threshold.

We should disambiguate our catch-all use of "fx" to better reflect what we are actually talking about. That will make fixes clearer and in the future help us to avoid this kind of situation. 

I know that this is a long-standing issue that has not originated with this PR, but I am a bit concerned that rather than making progress in the right direction, we are further entrenching the issue with more and more code that eventually should be changed.

Would it be possible to try to reflect better terminology in some of the variable and function names?

**Comment by zklaus** [comment]
Hm. I am not sure it's a good idea to mix those categories. Anyways, as far as I can see in this PR, we are really only dealing with cell measures, no? Perhaps then it is enough to refer to that, particularly in the test code.

**Comment by sloosvel** [comment]
I think it's convenient to have a general name to refer both ancillary variables and measure, because the steps of adding, checking and removing them from the cube are the same. In any case, even if the code changes, they are still going to be referred as `fx_variables` in the recipe. Maybe it would be better to leave it as it is for now and try to see if there is a better way to handle them in the upcoming months. As you said, this mix-up was not introduced in this PR.

File: tests/integration/preprocessor/_ancillary_vars/test_add_fx_variables.py [hunk 1]
```diff
@@ -169,6 +179,35 @@ def test_add_cell_measure_volume(self, tmp_path):
         cube = add_fx_variables(cube, fx_vars, CheckLevels.IGNORE)
         assert cube.cell_measure(self.fx_volume.standard_name) is not None
 
+    def test_clip_volume_timerange(self, tmp_path):
+        """Test timerange is clipped in time dependent measures."""
+        fx_vars = {
```
Comment (zklaus) [inline tests/integration/preprocessor/_ancillary_vars/test_add_fx_variables.py:184]:
```suggestion
        cell_measures = {
```
etc below

File: tests/integration/preprocessor/_ancillary_vars/test_add_fx_variables.py [hunk 2]
```diff
@@ -169,6 +179,35 @@ def test_add_cell_measure_volume(self, tmp_path):
...
+                'project': 'CMIP6',
+                'dataset': 'EC-Earth3',
+                'mip': 'Omon',
+                'frequency': 'mon',
+                'timerange': '195001/195003'}
+            }
+        self.monthly_volume.var_name = 'volcello'
+        self.monthly_volume.standard_name = 'ocean_volume'
+        self.monthly_volume.units = 'm3'
+        fx_file = str(tmp_path / 'volcello.nc')
```
Comment (zklaus) [inline tests/integration/preprocessor/_ancillary_vars/test_add_fx_variables.py:196]:
```suggestion
        cell_measure_file = str(tmp_path / 'volcello.nc')
```

Comment (zklaus) [reply tests/integration/preprocessor/_ancillary_vars/test_add_fx_variables.py:196]: etc below

**Review by zklaus — COMMENTED** [review]
I left a few comments to illustrate what I had in mind, i.e. not to change all the code that you basically didn't touch here, but rather to slightly nudge things in the right direction by using proper terminology in new tests etc.

Tangentially worth noting https://github.com/ESMValGroup/ESMValCore/pull/1082#issuecomment-1141787826 and the associated branch https://github.com/ESMValGroup/ESMValCore/compare/main...split-recipe

**Comment by sloosvel** [comment]
>Tangentially worth noting https://github.com/ESMValGroup/ESMValCore/pull/1082#issuecomment-1141787826 and the associated branch https://github.com/ESMValGroup/ESMValCore/compare/main...split-recipe

I guess this can be closed then?

**Comment by zklaus** [comment]
I'm not sure. @bouweandela, what do you think?

**Comment by zklaus** [comment]
Alright, then I suggest we forget all my comments, just get this in now as a stop-gap measure and hope that your reshuffle improves the overall situation either for 2.6.0 or shortly thereafter.

**Comment by bouweandela** [comment]
It looks like I won't be able to get #1609 done in time, so please go ahead and include this in v2.6 if you need it in that release.

**Comment by schlunma** [comment]
I will have a look on this on Friday or early next week.

**Review by schlunma — APPROVED** [review]
Code looks good!

Successfully tested this with the following recipe (took forever to run though, I'm really looking forward to https://github.com/ESMValGroup/ESMValCore/pull/1545):

```yaml
documentation:
  description: Test

  title: Test

  authors:
    - schlund_manuel


preprocessors:

  area_stat:
    area_statistics:
      operator: mean
      fx_variables:
        areacella:

  vol_stat:
    volume_statistics:
      operator: mean
      fx_variables:
        volcello:
          mip: Omon


diagnostics:

  test:

    variables:
      tas:
        preprocessor: area_stat
        mip: Amon
        exp: historical
        start_year: 2000
        end_year: 2014
      thetao:
        preprocessor: vol_stat
        mip: Omon
        exp: historical
        timerange: '200001/200001'
      additional_datasets:
        - {dataset: ACCESS-ESM1-5, project: CMIP6, ensemble: r1i1p1f1, grid: gn}
    scripts:
      null
```

In the current `main`, this recipe fails. Thanks @sloosvel!!

**Pull request merged**